### PR TITLE
docs(example): DraggableBox does not follow the cursor smoothly

### PR DESCRIPTION
**Pull request by ksokolovskyi**
# Description

This PR makes the `DraggableBox` movement more smooth and predictable by:
1) adjusting `FixtureDef` and `MouseJointDef` parameters
2) setting the `MouseJoint` target to `game.screenToWorld(event.deviceEndPosition)` rather than `event.localEndPosition`

Fixes https://github.com/flame-engine/flame/issues/3203

The comparison recordings are presented below:

<details closed><summary>GearJointExample</summary>

  | BEFORE | AFTER |
  |----------|----------|
  | <video src="https://github.com/flame-engine/flame/assets/30288967/8188ed18-e0aa-4c0f-a6c4-092015ba471c"> | <video src="https://github.com/flame-engine/flame/assets/30288967/2388bf23-ab32-43a8-af88-a974687b6901"> |

</details>

<details closed><summary>PrismaticJointExample</summary>

  | BEFORE | AFTER |
  |----------|----------|
  | <video src="https://github.com/flame-engine/flame/assets/30288967/80ca56cd-7b7c-4a12-a551-37f268011bb0"> | <video src="https://github.com/flame-engine/flame/assets/30288967/aaae0f7d-230b-47db-a1c2-cc39cace74e7"> |

</details>

<details closed><summary>PulleyJointExample</summary>

  | BEFORE | AFTER |
  |----------|----------|
  | <video src="https://github.com/flame-engine/flame/assets/30288967/1d40b0a0-57d2-444e-a324-63112548f9e3"> | <video src="https://github.com/flame-engine/flame/assets/30288967/6deb1226-0f96-440f-ba72-14806612bd32"> |

</details>

<details closed><summary>RopeJointExample</summary>

  | BEFORE | AFTER |
  |----------|----------|
  | <video src="https://github.com/flame-engine/flame/assets/30288967/60b2ceee-4a41-4443-a2db-de5515270f18"> | <video src="https://github.com/flame-engine/flame/assets/30288967/c06e35ff-2c59-4a02-b6bb-7611817a76c0"> |

</details>

## Checklist

- [X] I have followed the [Contributor Guide] when preparing my PR.
- [ ] I have updated/added tests for ALL new/updated/fixed functionality.
- [ ] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [X] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [X] No, this PR is not a breaking change.

## Related Issues

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

File: examples/lib/stories/bridge_libraries/flame_forge2d/utils/boxes.dart [hunk 1]
```diff
@@ -51,6 +58,7 @@ class DraggableBox extends Box with DragCallbacks {
     required super.startPosition,
     required super.width,
     required super.height,
+    super.density = 1000,
```
Comment (spydon) [inline examples/lib/stories/bridge_libraries/flame_forge2d/utils/boxes.dart:61]: How come it needs a density of 1000, that seems a bit excessive?

**Review by spydon — APPROVED** [review]
Lgtm, thanks for your contribution!
Just one small comment.

File: examples/lib/stories/bridge_libraries/flame_forge2d/utils/boxes.dart (hunk 1 above)
Comment (ksokolovskyi) [reply examples/lib/stories/bridge_libraries/flame_forge2d/utils/boxes.dart]: Thanks for the review. I set it to this value to decrease the oscillation, but after your comment, I tried to increase the `maxForce` and it helped as well. So I reverted the density to 10.

What do you think about the latest commits?

Comment (spydon) [reply examples/lib/stories/bridge_libraries/flame_forge2d/utils/boxes.dart]: Much better :+1:

**Review by spydon — APPROVED** [review]

**Comment by ksokolovskyi** [comment]
@spydon thanks for the review and merge!

**Pull request merged**